pkg/kubefed2/unjoin.go

```diff
@@ -167,10 +172,15 @@ func UnjoinCluster(hostConfig, clusterConfig *rest.Config, federationNamespace,
 		return err
 	}
 
-	clusterClientset, err := util.ClusterClientset(clusterConfig)
-	if err != nil {
-		glog.V(2).Infof("Failed to get unjoining cluster clientset: %v", err)
-		return err
+	var clusterClientset *client.Clientset = nil
+	if clusterConfig != nil {
+		clusterClientset, err = util.ClusterClientset(clusterConfig)
```

Comment:
Please check the definition of `ClusterClientSet`.  Since it calls `client.NewForConfigOrDie`, it will never return an error and will instead panic which is not what the code here expects.  I'm not sure why an error is being returned, I'm guessing this suboptimal code was inherited from fedv1.